platform/darwin/src/MGLStyle.mm

```diff
@@ -308,7 +308,7 @@ - (void)removeObjectFromLayersAtIndex:(NSUInteger)index
         [NSException raise:NSRangeException
```

Comment:
These off-by-one errors are confusing but it's still possible to remove the topmost layer.
Perhaps a greater than or equal to operator would make it easier to reason about?